src/transformers/models/gptj/modeling_gptj.py

```diff
@@ -55,9 +55,9 @@
 ]
 
 
-def create_sinusoidal_positions(num_pos: int, dim: int) -> torch.Tensor:
+def create_sinusoidal_positions(num_pos: int, dim: int, dtype: torch.dtype) -> torch.Tensor:
     inv_freq = 1.0 / (10000 ** (torch.arange(0, dim, 2) / dim))
-    sinusoid_inp = torch.einsum("i , j -> i j", torch.arange(num_pos, dtype=torch.float), inv_freq).float()
+    sinusoid_inp = torch.einsum("i , j -> i j", torch.arange(num_pos, dtype=torch.float), inv_freq).to(dtype)
```

Comment:
I think I specifically kept this for BC actually 😓 The original model does the same and the computation is not the same no?

Reply:
Will double check and keep you up